crates/core/src/masp.rs

```diff
@@ -238,6 +238,39 @@ pub fn encode_asset_type(
     .encode()
 }
 
+/// Encode the assets that are used for masp rewards
+pub fn encode_reward_asset_types(
+    native_token: &Address,
+) -> Result<[AssetType; 4], std::io::Error> {
+    use crate::token::{MaspDigitPos, NATIVE_MAX_DECIMAL_PLACES};
+    Ok([
+        encode_asset_type(
+            native_token.clone(),
+            NATIVE_MAX_DECIMAL_PLACES.into(),
+            MaspDigitPos::Zero,
+            Some(MaspEpoch::zero()),
+        )?,
```

Comment:
all of these will be infallible, anyway, no?

Reply:
The code for encoding is actually non-trivial involving some rejection sampling in the MASP crate, so it's not immediately clear that this cannot happen. That being said, I was trying to make the signature of `encode_reward_asset_types` consistent with `encode_asset_type`, and follow the precedent already set of propagating asset type encoding errors.